### ISSUE TITLE
Tooltip swallows onClick events on any of its children

### ISSUE THREAD

**Issue by markuso** [tleunen/react-mdl#65]
I am running into an issue with the `Tooltip` component in the latest release v0.14.0 where if there is an onClick handler on the child element it gets ignored like the tooltip is preventing propagation or something else.

Consider this sample code where the onClick never fires:

```jsx
<Tooltip label="Delete this item">
  <IconButton name="delete" onClick={() => console.log('TEST')} />
</Tooltip>
```

Could it be the new `MDLComponent` or how the `mdlUpgrade` is upgrading the tooltip which breaks any child events?

Before I upgraded, the tooltip never worked but it did not break the child events, but now it is the opposite. I was happy to see that the Tooltip works now, but had to rip it out as it broke the apps functionality. I am hoping there is a fix for this. Let me know your thoughts.

**Comment by faergeek** [comment]
It's an upstream issue
https://github.com/google/material-design-lite/issues/1196
https://github.com/google/material-design-lite/issues/1461

React uses event delegation for all events.
As a workaround you can try this out https://www.npmjs.com/package/react-native-listener
Or just use `addEventListener` on mount and `removeEventListener` on unmount manually.

**Comment by faergeek** [comment]
@tleunen I think we should fix it by binding events internally.

**Comment by faergeek** [comment]
Also @markuso did it automatically update from `0.13` to `0.14`?

**Comment by faergeek** [comment]
After some investigation I'm not sure if we can fix it without patching upstream or remove event handler and lose tooltip functionality on mobile (also seems like it doesn't work right on Mobile Safari anyway).

**Comment by markuso** [comment]
@faergeek No, react-mdl did not automatically upgrade from 0.13 to 0.14 but I needed to upgrade to get some of the latest fixes and changes as I needed them on the project. Also, to actually fix the Tooltip too since it was not functioning at all. Thank you for shedding some more light on this.

**Comment by tleunen** [comment]
Wow...
Yep the fact they use `event.stopPropagation();` is a bit problematic here... Maybe we can manually bind the click event for the element inside the tooltip?

Once the element is mounted, we manually add the click event (if one was provided), do you think it could work @faergeek ?

**Comment by faergeek** [comment]
In that case we should walk entire component tree under tooltip or just use event delegation :-)

**Comment by tleunen** [comment]
Why the entire tree? The main element linked to the tooltip should be enough.

**Comment by faergeek** [comment]
And what if there is more than one element inside?

**Comment by tleunen** [comment]
A tooltip can only have 1 single child, and I think the issue is there because MDL add an event on the linked child with the stopPropagation. That's why we should be fine by just manually mapping the event on that linked element.

**Comment by faergeek** [comment]
Even if we use event delegation, it's more safe to assume that there can be elements inside that single child and we should collect click handlers on them and call them with right `this`.

I started thinking that it's not so good solution as seemed at first.

Now I think it would be better to patch upstream again.

**Comment by tleunen** [comment]
I don't want to create a patch. The patch file is going to disappear once MDL goes 2.0.

They said in one of the issue you linked it's the normal behavior (for them), and they won't fix it (for now).
I know it would be better to fix it upstream, but it seems there are some reason why they do that.

**Comment by faergeek** [comment]
The reason is that it accidentally works on mobile (not in iOS Simulator, though) and seems like no one wants to dig into it and try to understand why and how it works :-) I tried myself.

On the other side I don't know why they use JS to just show hint.

**Comment by tleunen** [comment]
@faergeek any update? Is it fixed upstream?

**Comment by faergeek** [comment]
It's merged, but I'm not sure that it's released already, need to take a closer look.

**Comment by faergeek** [comment]
Looks like it should be in `1.0.6`

**Comment by tleunen** [comment]
I think it's fixed in 1.1, not in 1.0.6

**Comment by faergeek** [comment]
Confirming, it's not fixed in 1.0.6

**Comment by michaelguild13** [comment]
Here is an example for a simple fix for the onclick with mdl and react
https://jsfiddle.net/7k3upx7c/13/

**Comment by tleunen** [comment]
The problem is that it would required to be put inside every "clickable" component :/

**Comment by tleunen** [comment]
The issue is already fixed upstream in 1.1.x. So it will be included in react-mdl 1.4.0.
